src/creature.js

```diff
@@ -1934,4 +1934,39 @@ export class Creature {
 
 		this.game.signals.creature.dispatch('frozen', { creature: this, cryostasis });
 	}
+
+	get fatigueText() {
+		let text;
+		if (this.isFrozen()) {
+			text = this.isInCryostasis() ? 'Cryostasis' : 'Frozen';
+		} else if (this.isDizzy()) {
+			text = 'Dizzy';
+		} else if (this.materializationSickness) {
+			text = 'Sickened';
+		} else if (this.protectedFromFatigue || this.stats.fatigueImmunity) {
+			text = 'Protected';
+		} else if (this.isFragile()) {
+			text = 'Fragile';
+			// Display message if the creature has first become fragile
+
+			// TODO: This isn't necessarily the moment the creature has
+			// become fragile. The code will run twice if, e.g.,
+			// the creature is fragile, then fragile and dizzy, then fragile
```

Comment:
What status is "dizzy"?

Reply:
It looks like it's a status that can be set by a stomper:

https://github.com/FreezingMoon/AncientBeast/blob/893e174728e0b86a374738147a96ad4f68525842/src/abilities/Stomper.js#L514

Fwiw, `dizzy` wasn't added by this refactor. It was present in the code before the refactor:

https://github.com/FreezingMoon/AncientBeast/commit/fe35ee8ae8411a7ad9cb82a83e4aa450fbe37887?diff=split#diff-7a4b8145c1bf2bf544d29967ab26fcd9f62cfe7944aa1c0a1b255c8cdd41e001L2571

But even without `dizzy`, the logging would have a bug, it seems:

* creature is fragile
* any if-branches before fragile check become true
* if-branches before fragile check become false
* `this.game.log('%CreatureName' + this.id + '% has become fragile');`

It seems to me that – ideally – a check would happen whenever `this.stats.endurance` is set, and log if it's changed to 1 – the condition for `isFragile()`.

Reply:
I submitted a fix to #2175. 

To me, this would ideally be addressed with architecture. But it would mean making rather large changes to the codebase.

As it stands, a creature making an attack needs to call ...

https://github.com/FreezingMoon/AncientBeast/blob/893e174728e0b86a374738147a96ad4f68525842/src/abilities/Headless.js#L89

... if the result of the attack could potentially result in an update. 

Ideally, that shouldn't be necessary. [There's more discussion in the PR.](https://github.com/FreezingMoon/AncientBeast/pull/2177)